core/controller/src/main/scala/whisk/core/loadBalancer/LoadBalancerData.scala

```diff
@@ -17,107 +17,130 @@
+   * Get the number of activations for each invoker.
    *
-   * @param invoker The invoker to get the activation count for
    * @return a map (invoker -> number of activations queued for the invoker)
    */
-  def activationCountOn(invoker: InstanceId): Int = {
-    activationByInvoker.get(invoker).map(_.get).getOrElse(0)
-  }
+  def activationCountPerInvoker: Future[Map[String, Int]]
 
   /**
    * Get an activation entry for a given activation id.
    *
    * @param activationId activation id to get data for
    * @return the respective activation or None if it doesn't exist
    */
-  def activationById(activationId: ActivationId): Option[ActivationEntry] = {
-    activationsById.get(activationId)
-  }
+  def activationById(activationId: ActivationId): Option[ActivationEntry]
 
   /**
    * Adds an activation entry.
    *
-   * @param id identifier to deduplicate the entry
+   * @param id     identifier to deduplicate the entry
    * @param update block calculating the entry to add.
    *               Note: This is evaluated iff the entry
    *               didn't exist before.
    * @return the entry calculated by the block or iff it did
    *         exist before the entry from the state
    */
-  def putActivation(id: ActivationId, update: => ActivationEntry): ActivationEntry = {
-    activationsById.getOrElseUpdate(id, {
-      val entry = update
-      totalActivations.incrementAndGet()
-      activationByNamespaceId.getOrElseUpdate(entry.namespaceId, new AtomicInteger(0)).incrementAndGet()
-      activationByInvoker.getOrElseUpdate(entry.invokerName, new AtomicInteger(0)).incrementAndGet()
-      entry
-    })
-  }
+  def putActivation(id: ActivationId, update: => ActivationEntry): ActivationEntry
 
   /**
    * Removes the given entry.
    *
    * @param entry the entry to remove
    * @return The deleted entry or None if nothing got deleted
    */
-  def removeActivation(entry: ActivationEntry): Option[ActivationEntry] = {
-    activationsById.remove(entry.id).map { x =>
-      totalActivations.decrementAndGet()
-      activationByNamespaceId.getOrElseUpdate(entry.namespaceId, new AtomicInteger(0)).decrementAndGet()
-      activationByInvoker.getOrElseUpdate(entry.invokerName, new AtomicInteger(0)).decrementAndGet()
-      x
-    }
-  }
+  def removeActivation(entry: ActivationEntry): Option[ActivationEntry]
 
   /**
    * Removes the activation identified by the given activation id.
    *
    * @param aid activation id to remove
    * @return The deleted entry or None if nothing got deleted
    */
+  def removeActivation(aid: ActivationId): Option[ActivationEntry]
+}
+
+/**
+ * Encapsulates data used for loadbalancer and active-ack bookkeeping.
+ *
+ * Note: The state keeping is backed by distributed akka actors. All CRUDs operations are done on local values, thus
+ * a stale value might be read.
+ */
+class LoadBalancerData(implicit val actorSystem: ActorSystem) extends BookkeepingData {
```

Comment:
Can we encapsulate this in its own file and name it `LoadBalancerDataDistributed`. (Or adopt the more common naming convention "specialization attribute up front" like "DistributedLoadBalancerData")